test/functional/ui/float_spec.lua

```diff
@@ -104,14 +104,20 @@ describe('float window', function()
   end)
 
   it('open with WinNew autocmd', function()
-    local res = exec_lua([[
-      local triggerd = false
+    local new_triggered_before_enter, new_curwin, win = unpack(exec_lua([[
+      local enter_triggered = false
+      local new_triggered_before_enter = false
+      local new_curwin
       local buf = vim.api.nvim_create_buf(true, true)
+      vim.api.nvim_create_autocmd('WinEnter', {
+        callback = function()
+          enter_triggered = true
+        end
+      })
       vim.api.nvim_create_autocmd('WinNew', {
-        callback = function(opt)
-          if opt.buf == buf then
-            triggerd = true
-          end
+        callback = function()
+          new_triggered_before_enter = not enter_triggered
+          new_curwin = vim.api.nvim_get_current_win()
```

Comment:
Nice spotting! I made a few mistakes in these test changes which you've fixed. :+1:

Reply:
Yeah haha noticed that the test was passing but the variable was never being set, so I looked into it.